src/services/api-service.tsx

```diff
@@ -0,0 +1,100 @@
+import { IProvider } from "../interfaces/iprovider";
+import { Observable } from "rxjs";
+import { Weather } from "../models/Weather";
+import { ApiInfo } from './api-info';
+import { Coordinate } from "../models/Coordinate";
+import { HourlyForecast } from "../models/HourlyForecast";
+/*
+  The purpose of this class is to be the main point of API consume.
+  This class will be used as the Service that is allowed to request using REST to OpenWeather API
+ */
+export class ApiService {
+
+    ApiInfo: ApiInfo = new ApiInfo();
+
+    constructor(private provider: IProvider) { }
+  
+    /**
+     * This method returns the weather for the given city
+     * @param cityName Name of the city
+     */
+    getCurrentWeather(cityName: string): Observable<Weather> {
+      return Observable.create((observer: { next: (arg0: any) => void; }) => {
```

Comment:
What is `arg0` meaning in this context?

Reply:
It was a Lint suggestion because "observer" didn't had an explicit type.